packages/solid-urql/src/createQuery.ts

```diff
@@ -0,0 +1,334 @@
+    operation: operation,
+    fetching: false,
+    data: undefined,
+    error: undefined,
+    extensions: undefined,
+    hasNext: false,
+    stale: false,
+  };
+
+  const [result, setResult] =
+    createStore<CreateQueryState<Data, Variables>>(initialResult);
+
+  // `createResource` is only way to trigger suspense, suspense handling works as follows:
+  // 1. Calling `dataResource` mounts resource to closes suspense. Thus depending on where it is called
+  // it could mount in same component or parent. For usage convince we call it by proxy when using `data`
+  // 2. After any changes we call `refetch` which results in creation of new promise and depending on `fetching`
+  // value we either discard promise without resolving or resolve it. Discard doesn't result in anything while
+  // resolve finishes suspense
+  const [dataResource, { refetch, mutate }] = createResource<
+    CreateQueryState<Data, Variables>
+  >(() => {
+    return new Promise(resolve => {
+      const got = unwrap(result);
+      if (got.fetching) {
+        return;
+      }
+
+      resolve(got);
+    });
+  });
+
+  const sub = pipe(
+    resultSourceSubject.source,
+    switchMap(subscription$ => {
+      // On pause source is set to undefined
+      if (subscription$ === undefined) {
+        return fromValue({ fetching: false });
+      }
+
+      return concat([
+        // Fetch start state
+        fromValue({ fetching: true, stale: false }),
+        pipe(
+          subscription$,
+          map(({ stale, data, error, extensions, operation }) => ({
+            fetching: false,
+            stale: !!stale,
+            data,
+            error,
+            operation,
+            extensions,
+          }))
+        ),
+        // Fetch finish state
+        fromValue({ fetching: false, stale: false }),
+      ]);
+    }),
+    // Merging partial results into previous
+    scan(
+      (result: CreateQueryState<Data, Variables>, partial: any) => ({
+        ...result,
+        ...partial,
+      }),
+      initialResult
+    ),
+    subscribe(result => {
+      batch(() => {
+        setResult(result);
+        mutate(result);
+        refetch();
+      });
+    })
+  );
+
+  createComputed(() => {
+    const pause = getPause();
+    if (pause === true) {
+      resultSourceSubject.next(undefined);
+      return;
+    }
+
+    const request = createRequest(args.query, getVariables() as any);
+    const context: Partial<OperationContext> = {
+      requestPolicy: getRequestPolicy(),
+      ...getContext(),
+    };
+
+    resultSourceSubject.next(client.executeQuery(request, context));
+  });
+
+  onCleanup(() => sub.unsubscribe());
+
+  batch(() => {
+    mutate(() => unwrap(result));
+    refetch();
+  });
+
+  const executeQuery: CreateQueryExecute = opts => {
+    const request = createRequest(args.query, getVariables() as any);
+    const context: Partial<OperationContext> = {
+      requestPolicy: opts?.requestPolicy ?? getRequestPolicy(),
```

Comment:
There's a couple of partially stylistic, partially transpilation/browser-support related issues in this PR. Instead of listing them out, I merged a PR into `main`. So, once you rebase it'll tell you about them automatically: https://github.com/urql-graphql/urql/pull/3350

Reply:
Fixed, didn't notice that I had invalid line endings git config on my current station...